langlib/langlib-test/src/test/resources/test-src/valuelib_test.bal

```diff
@@ -3671,8 +3671,9 @@ function testCloneWithTypeToTableNegative() {
     UnionTable2|error t3 = tab2.cloneWithType();
     if (t3 is error) {
         assertEquality("{ballerina/lang.value}ConversionError", t3.message());
-        assertEquality("'map<json>[]' value cannot be converted to 'UnionTable2': " + errMsgSuffix,
-        <string> checkpanic t3.detail()["message"]);
+        // TODO: fix #41288
+        assertEquality("'map<json>[]' value cannot be converted to '((Table1 & readonly)|(Table2 & readonly))': " +
```

Comment:
I don't think we can go ahead with this as a known issue, given that it can break existing code.

cc @gimantha

Reply:
Fixed with [7b105fd](https://github.com/ballerina-platform/ballerina-lang/pull/41286/commits/7b105fd7de79dc95c88d0e00f7cdf82ccb860daa)